pkg/apis/storage/types.go

```diff
@@ -310,6 +327,36 @@ type CSIDriverSpec struct {
 	VolumeLifecycleModes []VolumeLifecycleMode
 }
 
+// FSGroupPolicy specifies if a CSI Driver supports modifying
+// volume ownership and permissions of the volume to be mounted.
+// More modes may be added in the future.
+type FSGroupPolicy string
+
+const (
+	// Heuristic indicates that each volume should be examined
```

Comment:
Sorry I'm flip flopping on naming since we have the other discussion that that was moving away from the "Heuristic" term.

What about something like "ReadWriteOnce" or "ReadWriteOnceOnly"?

cc @gnufied @jsafrane

Reply:
+1 to ReadWriteOnce

ReadWriteOnceOnly would imply that the PV has to be [RWO] only, while fsGroup is applied also for PVs with [RWO, RWX]

Reply:
The way that I understand the current code - we apply the fsGroup if the following criteria are met:

* fstype is specified
* the volume is not readonly
* the access mode contains ReadWriteOnce

So if the accessMode contains RWX and RWO, then we'll still apply the defined fsGroup. This makes me a bit hesitant to give it a name that includes ReadWriteOnce. 

Would any of the following work:

* Legacy - has the issue of not being clear on what the behaviour is.
* Automatic - doesn't explicitly say what it does, but we can describe the logic behind it, similar to what we currently have defined for Heuristic.
* AttemptOnMount - a bit more explicit than the above, but a similar thought.

Reply:
yes `ReadWriteOnce` does not capture current behaviour. I liked "Evaluate" or even "Automatic"...

Reply:
Based on the call today, we're proposing the following:

* File - the driver supports fsGroup (block based)
* None -  the driver does not support fsGroup (such as NFS with root-squash)
* ReadWriteOnceWithFSType - current/legacy behavior, requires RWO in access modes and fstype defined.

Additional names proposed for ReadWriteOnceWithFSType were:

* FileLegacy
* FileAutomatic
* FileAutodetect
* Autodetect
* FallbackAutodetect
* FileRWO
* RWOFSType
* FileHeuristic